app/components/mandatarissen/fractie-selector.js

```diff
@@ -29,18 +29,64 @@ export default class MandatenbeheerFractieSelectorComponent extends Component {
   }
 
   async loadFracties() {
-    let fracties = await this.fetchFracties();
+    let fracties = [];
+
+    if (this.isUpdatingFractie && this._fractie) {
+      fracties = await this.getFractiesWhenUpdateState();
+    } else {
+      fracties = await this.fetchFracties();
+    }
+
+    let onafhankelijke = await this.getIndependentFractie(fracties);
 
-    let onafhankelijke = fracties.find((f) =>
-      f.get('fractietype.isOnafhankelijk')
-    );
     if (!onafhankelijke) {
       onafhankelijke = await this.createOnafhankelijkeFractie();
-      fracties = [...fracties, onafhankelijke];
+      this.fractieOptions = [...fracties, onafhankelijke];
+      return;
     }
+
     this.fractieOptions = fracties;
   }
 
+  async getFractiesWhenUpdateState() {
+    const mandataries = await this.store.query('mandataris', {
```

Comment:
I find mandataries really confusing. Maybe better to use mandatarissen or maybe mandatees?

Reply:
Updated the naming